Frends.Community.Email.Tests/ExchangeEmailTests.cs

```diff
@@ -0,0 +1,651 @@
+            await SendTestEmailWithAttachment(subject, "OnlyAttachment.txt");
+            await SendTestEmail(subject);
+
+            var settings = new ExchangeSettings
+            {
+                TenantId = _tenantID,
+                AppId = _applicationID,
+                Username = _username,
+                Password = _password,
+                Mailbox = _mailbox
+            };
+            var options = new ExchangeOptions
+            {
+                MaxEmails = 2,
+                DeleteReadEmails = false,
+                GetOnlyUnreadEmails = false,
+                MarkEmailsAsRead = false,
+                IgnoreAttachments = false,
+                GetOnlyEmailsWithAttachments = true,
+                AttachmentSaveDirectory = dirPath,
+                EmailSubjectFilter = subject
+            };
+
+            Directory.CreateDirectory(dirPath);
+            var result = await ReadEmailTask.ReadEmailFromExchangeServer(settings, options, new CancellationToken());
+            Assert.IsTrue(result.Count == 1);
+            Directory.Delete(dirPath, true);
+            await DeleteMessages(subject);
+        }
+
+        [Test]
+        public async Task ReadEmailFromExchangeServer_ShouldOverwriteAttachmentsTest()
+        {
+            var dirPath = Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "../../../OverwriteData/");
+            var subject = "Overwrite Attchment Test";
+            await SendTestEmailWithAttachment(subject, "OverwriteAttachment.txt");
+            var settings = new ExchangeSettings
+            {
+                TenantId = _tenantID,
+                AppId = _applicationID,
+                Username = _username,
+                Password = _password,
+                Mailbox = _mailbox
+            };
+            var options = new ExchangeOptions
+            {
+                MaxEmails = 1,
+                DeleteReadEmails = false,
+                GetOnlyUnreadEmails = false,
+                MarkEmailsAsRead = false,
+                IgnoreAttachments = false,
+                GetOnlyEmailsWithAttachments = true,
+                AttachmentSaveDirectory = dirPath,
+                OverwriteAttachment = true
+            };
+
+            Directory.CreateDirectory(dirPath);
+            var result = await ReadEmailTask.ReadEmailFromExchangeServer(settings, options, new CancellationToken());
+            Assert.IsTrue(File.Exists(result[0].AttachmentSaveDirs[0]));
+            Assert.IsTrue(Directory.GetFiles(dirPath).Length == 1);
+            result = await ReadEmailTask.ReadEmailFromExchangeServer(settings, options, new CancellationToken());
+            Assert.IsTrue(File.Exists(result[0].AttachmentSaveDirs[0]));
+            Assert.IsTrue(Directory.GetFiles(dirPath).Length == 1);
+            Directory.Delete(dirPath, true);
+            await DeleteMessages(subject);
+        }
+
+        [Test]
+        public async Task SendEmailWithPlainTextTest()
```

Comment:
Sudden test naming change? :) 

Also, would help to split up test files by task. This way you can again move the general stuff like general input/server config to [SetUp] and this way clear up a lot of test code.